src/sheldon.erl

```diff
@@ -59,17 +59,24 @@ check_words(Words, Config) -> check_words(Words, [], Config).
                  , sheldon_config:config()
                  ) -> [string()].
 check_words([], Acc, _Config) -> Acc;
-check_words([Word | Rest], Acc, Config = #{lang := Lang}) ->
+check_words([Word | Rest], Acc, Config) ->
   Normalized = sheldon_utils:normalize(Word),
-  Acc1 = case is_correct(Lang, Normalized) of
+  Acc1 = case correctly_spelled(Normalized, Config) of
            true  -> Acc;
            false -> [Normalized | Acc]
          end,
   check_words(Rest, Acc1, Config).
 
--spec is_correct(sheldon_dictionary:language(), string()) -> boolean().
-is_correct(Language, Word) ->
-  case sheldon_utils:is_number(Word) of
-    false -> sheldon_dictionary:exist_in_dictionary(Language, Word);
+-spec correctly_spelled(string(), sheldon_config:config()) -> boolean().
+correctly_spelled(Word, Config = #{lang := Lang}) ->
+  case sheldon_utils:is_number(Word) or ignore(Word, Config) of
+    false -> sheldon_dictionary:exist(Word, Lang);
     true  -> true
   end.
+
+-spec ignore(string(), sheldon_config:config()) -> boolean().
+ignore(Word, _Config = #{ ignore_words    := IgnoredWords
+                        , ignore_patterns := Patterns
+                        }) ->
+  lists:member(string:to_lower(Word), IgnoredWords)
+    or sheldon_utils:match_in_patterns(Word, Patterns).
```

Comment:
Same here, change `or` to `orelse` to avoid calling `sheldon_utils:match_in_patterns/2` for ignored words.